crates/cheatnet/src/execution/syscalls.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::Result;
+use blockifier::execution::{
+    execution_utils::felt_to_stark_felt, syscalls::hint_processor::SyscallHintProcessor,
+};
+use cairo_felt::Felt252;
+use cairo_lang_casm::operand::ResOperand;
+use cairo_lang_casm::operand::{BinOpOperand, DerefOrImmediate, Operation, Register};
+use cairo_lang_runner::casm_run::{extract_relocatable, vm_get_range};
+use cairo_lang_utils::bigint::BigIntAsHex;
+use cairo_vm::types::relocatable::Relocatable;
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use cairo_vm::vm::{errors::hint_errors::HintError, vm_core::VirtualMachine};
+use starknet_api::hash::StarkFelt;
+
+pub fn get_syscall_selector(
+    syscall: &ResOperand,
+    vm: &mut VirtualMachine,
+    syscall_handler: &SyscallHintProcessor<'_>,
+) -> Result<StarkFelt, HintError> {
+    let initial_syscall_ptr = get_ptr_from_res_operand_unchecked(vm, syscall);
+    syscall_handler.verify_syscall_ptr(initial_syscall_ptr)?;
+    let selector = stark_felt_from_ptr_immutable(vm, &initial_syscall_ptr)?;
+    Ok(selector)
+}
+
+pub fn stark_felt_from_ptr_immutable(
+    vm: &VirtualMachine,
+    ptr: &Relocatable,
+) -> Result<StarkFelt, VirtualMachineError> {
+    Ok(felt_to_stark_felt(&felt_from_ptr_immutable(vm, ptr)?))
```

Comment:
We can use our conversions trait here. Can we get rid of those chain call entirely here?